snowdrop-block/src/Snowdrop/Block/Chain.hs

```diff
@@ -15,15 +15,15 @@ import           Snowdrop.Block.StateConfiguration (BlkStateConfiguration (..))
 import           Snowdrop.Block.Types (Block (..), BlockHeader, BlockRef, BlockUndo, Blund (..),
                                        CurrentBlockRef (..), PrevBlockRef (..), RawBlund,
                                        RawPayload)
-import           Snowdrop.Core (ChgAccum, ERoComp, QueryERo, queryOne)
+import           Snowdrop.Core (ChgAccum, ERoCompU, QueryERo, convertEffect, queryOne)
 import           Snowdrop.Util (NewestFirst (..), OldestFirst (..), newestFirstFContainer,
                                 toOldestFirst)
 
 -- | Load up to @maxDepth@ blocks from the currently adopted block sequence.
 iterateChain
     :: forall blkType conf xs m .
     ( QueryERo xs (BlundComponent blkType)
-    , m ~ ERoComp conf xs
+    , m ~ ERoCompU conf xs
```

Comment:
Here and later `ERoCompU` is not actually needed. Try to use `convertEffect` on the caller site.

Reply:
I do not like this more stronger type than necessary either. 

But the problem is that on caller site all I have is `BlkStateConfiguration (ChgAccum conf) blkType (ERoCompU conf (DbComponents conf))` which we get by calling `inmemoryBlkStateConfiguration` function.

At this point I cannot satisfy constraint `m ~ ERoComp conf xs`, and `convertEffect` won't help in this case.